pkg/debugapi/chunk.go

```diff
@@ -63,3 +64,47 @@ func (s *Service) removeChunk(w http.ResponseWriter, r *http.Request) {
 	}
 	jsonhttp.OK(w, nil)
 }
+
+// chunkListAddresses lists all addresses related to the supplied one.
+func (s *Service) chunkListAddresses(w http.ResponseWriter, r *http.Request) {
+	addr, err := swarm.ParseHexAddress(mux.Vars(r)["address"])
+	if err != nil {
+		s.logger.Debugf("debug api: parse chunk address: %v", err)
+		jsonhttp.BadRequest(w, "bad address")
+		return
+	}
+
+	ctx := r.Context()
+
+	has, err := s.storer.Has(ctx, addr)
+	if err != nil {
+		s.logger.Debugf("debug api: localstore has: %v", err)
+		jsonhttp.BadRequest(w, err)
+		return
+	}
+	if !has {
+		jsonhttp.NotFound(w, nil)
+		return
+	}
+
+	var addresses []swarm.Address
+	iterFn := func(leaf swarm.Address) error {
+		chunk, err := s.storer.Get(ctx, storage.ModeGetRequest, leaf)
+		if err == nil || errors.Is(err, storage.ErrNotFound) {
+			if chunk != nil {
+				addresses = append(addresses, chunk.Address())
+			}
+			return nil
+		}
+		return err
+	}
+
+	if err := s.traverser.Traverse(ctx, addr, iterFn); err != nil {
+		s.logger.Debugf("debug api: traverse of %s failed: %v", addr, err)
+		s.logger.Error("debug api: traverse failed")
+		jsonhttp.InternalServerError(w, nil)
+		return
+	}
+
+	jsonhttp.OK(w, addresses)
```

Comment:
A JSON array is returned here as the response. I would suggest to wrap it inside a JSON object as it allows the response to be extended later without breaking changes.